File: algorithms/2108.find-first-palindromic-string-in-the-array/solution.rs
impl Solution {
    pub fn first_palindrome(words: Vec<String>) -> String {
        for word in words {
            if Solution::is_palindrome(&word) {
                return word;
            }
        }

        "".to_string()
    }

    fn is_palindrome(s: &str) -> bool {
        let chars: Vec<char> = s.chars().collect();
        let (mut left, mut right) = (0, chars.len() - 1);

        while left < right {
            if chars[left] != chars[right] {
                return false;
            }

            left += 1;
            right -= 1;
        }

        true
    }
}

File: algorithms/2108.find-first-palindromic-string-in-the-array/solution.go
func firstPalindrome(words []string) string {
	for _, word := range words {
		if isPalindrome(word) {
			return word
		}
	}

	return ""
}

func isPalindrome(word string) bool {
	r := []rune(word)
	left, right := 0, len(word)-1

	for left < right {
		if r[left] != r[right] {
			return false
		}

		left++
		right--
	}

	return true
}
